components/plugin_marketplace/marketplace_usage_message.tsx

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {PreferenceType} from '@mattermost/types/preferences';
+
+import {getCloudLimits} from 'mattermost-redux/selectors/entities/cloud';
+import {getIntegrationsUsage} from 'mattermost-redux/selectors/entities/usage';
+import {cloudFreeEnabled, getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CloseIcon from 'components/widgets/icons/close_icon';
+import CloudStartTrialButton from 'components/cloud_start_trial/cloud_start_trial_btn';
+
+const HIDE_USAGE_MESSAGE_PREFERENCE_CATEGORY = 'usage_limits';
+const HIDE_USAGE_MESSAGE_PREFERENCE_NAME = 'hide_marketplace_usage_message';
+
+const usePreferenceCheck = (): [boolean, (value: boolean) => void] => {
+    const userId = useSelector(getCurrentUserId);
+    const preferences = useSelector(getMyPreferences);
+
+    const key = getPreferenceKey(HIDE_USAGE_MESSAGE_PREFERENCE_CATEGORY, HIDE_USAGE_MESSAGE_PREFERENCE_NAME);
+    const preference = preferences[key];
+    const hiddenDueToPreference = preference?.value === 'true';
+
+    const dispatch = useDispatch();
+    const setHidePreference = (value: boolean) => {
+        const preference: PreferenceType = {
+            category: HIDE_USAGE_MESSAGE_PREFERENCE_CATEGORY,
+            name: HIDE_USAGE_MESSAGE_PREFERENCE_NAME,
+            user_id: userId,
+            value: value.toString(),
+        };
+        dispatch(savePreferences(userId, [preference]));
+    };
+
+    return [hiddenDueToPreference, setHidePreference];
+};
+
+export default function MarketplaceUsageMessage() {
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const integrationsUsage = useSelector(getIntegrationsUsage);
+    const limits = useSelector(getCloudLimits);
+    const [hiddenDueToPreferences, setHidePreference] = usePreferenceCheck();
+    const intl = useIntl();
+
+    if (hiddenDueToPreferences || !isCloudFreeEnabled || !integrationsUsage?.enabledLoaded || !limits.integrations?.enabled) {
+        return (
+            <button
+                className='btn btn-primary'
+                onClick={() => setHidePreference(false)}
+            >
+                {'DEBUG reset preference'}
+            </button>
+        );
+    }
```

Comment:
Should be removed

Reply:
@neallred Will the `limits.integrations?.enabled` piece already cover that case?

Reply:
ahh, it will if you do some sort of nil check to not render in that case, yeah. i.e. if instead of returning that button you return null.